### PR TITLE
Colors now stretch from the extremities of the viewport.

### PR DESCRIPTION
**Pull request by Saucistophe**
...And fixed lint errors.
Solves #94

### CODE REVIEW

File: lib/trianglify.js [hunk 1]
```diff
@@ -108,11 +107,11 @@ function Trianglify(opts) {
 
   // Set up normalizers
   var norm_x = function(x) {
-    return _map(x, [-bleed_x, width+bleed_x], [0, 1]);
+    return _map(x, [0, width], [0, 1]);
```
Comment (qrohlf) [inline lib/trianglify.js]: The normalization here previously returned a float strictly within the interval `[0, 1]` (inclusive), and it's possible that API consumers depend on this behavior in their `color_function` implementation.

With the changes here, we'll be passing numbers outside that range to the `color_function`: 

screen shot 2018-10-18 at 9 17 20 am

While `chroma.interpolate` is able to handle this edge case, a 3rd party `color_function` may not be.

I'm on-board with changing the normalization function to start and end the gradient in the viewport, but I'd like to clamp out-of-bounds values to 0 or 1. Here's a rough sketch of how that might look:

```
var norm_x = function(x) {
  return _clamp(
    _map(x, [0, width], [0, 1]),
    [0, 1]
  )
}

// and here's my suggested implementation for clamp() (should go right above the function definition for _map)
function _clamp (num, interval) {
  return Math.min(
    Math.max(num, interval[0]),
    interval[1]
  )
}
```

File: lib/trianglify.js [hunk 2]
```diff
@@ -82,9 +82,8 @@ function Trianglify(opts) {
         .scale(opts.x_colors || opts.y_colors)
         .mode(opts.color_space);
 
-      var gradient = opts.x_colors
-        ? function(x, y) { return scale(x) }
-        : function(x, y) { return scale(y) };
+      gradient = opts.x_colors ? function(x, y) { return scale(x); }
```
Comment (qrohlf) [inline lib/trianglify.js]: Any reason for this whitespace change?

File: lib/trianglify.js (hunk 1 above)
Comment (Saucistophe) [reply lib/trianglify.js]: Ooh bummer! You are totally right, I overlooked the details of _map, I assumed it was already the case.

File: lib/trianglify.js (hunk 2 above)
Comment (Saucistophe) [reply lib/trianglify.js]: When running _npm run start_, the linter complained!

Comment (qrohlf) [reply lib/trianglify.js]: oops, probably my bad. Sorry about that!

**Comment by qrohlf** [comment]
This looks good, going to go ahead and merge. 

Because this is a breaking change, it'll go out under the `v2.0.0` release. I should have time later this week to roll that out. Thanks for your contribution!

**Comment by Saucistophe** [comment]
Well, thank _you_ for taking the time and creating this awesome lib. Much appreciated!

**Pull request merged**